synapse_invite_checker/invite_checker.py

```diff
@@ -167,21 +174,24 @@ def __init__(self, config: InviteCheckerConfig, api: ModuleApi):
         ) as db_conn:
             self.store = InviteCheckerStore(api._store.db_pool, db_conn, api._hs)
 
-        InfoResource(self.config, self.__version__).register(self.resource)
-        ContactsResource(self.api, self.store, self.config).register(self.resource)
-        ContactResource(self.api, self.store, self.config).register(self.resource)
+        ContactManagementInfoResource(self.config, self._TMCM_schema_version).register(
+            self.resource
+        )
+        ContactsResource(self.api, self.store).register(self.resource)
+        ContactResource(self.api, self.store).register(self.resource)
 
-        self.api.register_web_resource(f"{config.api_prefix}", self.resource)
-        logger.info("Module initialized at %s", config.api_prefix)
+        MessengerInfoResource(self.config, self._TMI_schema_version).register(
+            self.resource
+        )
+
+        self.api.register_web_resource(BASE_API_PREFIX, self.resource)
+        logger.info("Module initialized at %s", BASE_API_PREFIX)
```

Comment:
can we do 2 register_web_resource calls instead to get rid of the BASE_API_PREFIX and provide a slightly clearer grouping for those resources. Not strictly required, but it confused me a bit.

Reply:
I wasn't sure you could do that, at least without having separate `JsonResource`s. I personally like having a common root, so everything would be mounted from the same call. At some point there was to be a migration of where the old contact management api is mounted, and letting all of it have a common root. I think I was going with lowest common denominator :grin: which would also have let us have that nice universal location for the `__version__` to be exposed

Reply:
I spent a little time digging through the utility code that makes this work. The bottom line is that yes, you can do that if you wish. It would require constructing a new `JsonResource` for each branch instead of attaching everything to a common trunk. 

Otherwise it tries to parse all the regex paths twice when setting everything up, which may lead to issues in the future if one gets missed. I would prefer to have a common trunk and just attach everything once

Reply:
Yes, that was my assumption. Have different json resources for the different APIs and one day we delete the one for the contact management API.

Reply:
That is handled by just removing the object altogether. It doesn't exist and therefore it won't be registered. Clean removal. Unless you are considering a hot-reload scenario, I think that's over kill

Reply:
After an out-of-band conversation about this, I understand better. I'm not completely sure it's warranted, as I think a comment pointing out which resources belong to which API would be just as good. It's 2 lines versus like 7 lines, not 2 vs 100. Let's leave these as-is for now, ok?

Reply:
comments added in [76ad966](https://github.com/famedly/synapse-invite-checker/pull/33/commits/76ad966466c81f0bf14d86c9fb01afcd0764bb72)